File: JS/core/optionex.js
import { some, value } from "../.fable/fable-library.3.0.0/Option.js";

export function call(f, opt) {
    if (opt == null) {
    }
    else {
        const x = value(opt);
        f(x);
    }
}

export function run(f, opt) {
    if (opt == null) {
    }
    else {
        const x = value(opt);
        f(x);
    }
}

export function runWhenNone(f, opt) {
    if (opt == null) {
        f();
    }
}

export function execute(f, opt) {
    if (opt == null) {
        return void 0;
    }
    else {
        const x = value(opt);
        return some(f(x));
    }
}

export function executeDefault(def, f, opt) {
    if (opt == null) {
        return def;
    }
    else {
        const x = value(opt);
        return f(x);
    }
}

export function calcDef(f, opt) {
    if (opt == null) {
        return f();
    }
    else {
        const x = value(opt);
        return x;
    }
}

File: src/core/optionex.fs
module Option

let call f opt =
  match opt with
  | Some x -> f x
  | None -> ()

let run f opt =
  match opt with
  | Some x -> f x
  | None -> ()

let runWhenNone f opt =
  match opt with
  | Some _ -> ()
  | None -> f ()

let execute f opt =
  match opt with
  | Some x -> Some <| f x
  | None -> None

let executeDefault def f opt =
  match opt with
  | Some x -> f x
  | None -> def

let calcDef f opt =
  match opt with
  | Some x -> x
  | None -> f ()
